proxy/ParentSelection.h

```diff
@@ -102,6 +102,7 @@ struct pRecord : ATSConsistentHashNode {
   const char *scheme; // for which parent matches (if any)
   int idx;
   float weight;
+  char hash_string[MAXDNAME + 1];
```

Comment:
Why not `std::string`?

Reply:
hash_string really isn't necessary as it is only used once to set the 'name' field in the precord.  I'll remove it

Reply:
on second thought, we need to store it in 'hash_string'.  I'm okay with this 'char'  if @SolidWallOfCode  is.

Reply:
on second thought, 'hash_string' is needed.  Also, I don't think that it should be a std::string.  It needs to be the same as 'hostname' . because ConsistentHash needs to use one or the other in the exact same way, setting pointer for building the hash ring.  If you make hash_string a std::string then you need to change 'hostname' to one as well.  Since 'hostname' is used elsewhere in the state machine, I think we shouldn't make hash_string a std::string.